src/main/java/com/mcmoddev/orespawn/json/OS1Reader.java

```diff
@@ -84,45 +98,46 @@ public static void loadEntries(Path confDir) {
 
 	}
 
-	private static void loadOre(JsonObject oElem, DimensionLogic dimensionLogic) {
-		String blockID = oElem.get("blockID").getAsString();
-		String modId = blockID.contains(":")?blockID.substring(0, blockID.indexOf(':')):"minecraft";
-		String name = blockID.contains(":")?blockID.substring(blockID.indexOf(':')+1):blockID;
-		int meta = oElem.has("metaData")?oElem.get("metaData").getAsInt():0;
-		ResourceLocation blockKey = new ResourceLocation(blockID);
-		String biome = "biomes";
-		
-		if( !Block.REGISTRY.containsKey(blockKey) ) {
-			OreSpawn.LOGGER.warn("Asked to spawn block "+modId+":"+name+" that does not exist");
-			return;
-		}
-		
-		Block b = Block.REGISTRY.getObject(blockKey);
-		@SuppressWarnings("deprecation")
-		IBlockState bs = meta==0?b.getDefaultState():b.getStateFromMeta(meta);
-		
-		float frequency = oElem.has("frequency")?oElem.get("frequency").getAsFloat():20.0f;
-		int size = oElem.has("size")?oElem.get("size").getAsInt():8;
-		int maxHeight = oElem.has("maxHeight")?oElem.get("maxHeight").getAsInt():255;
-		int minHeight = oElem.has("minHeight")?oElem.get("minHeight").getAsInt():0;
-		int variation = oElem.has("variation")?oElem.get("variation").getAsInt():(int)(0.5f * size);
-		List<Biome> biomes;
-		
-		if( oElem.has(biome) && oElem.get(biome).getAsJsonArray().size() > 0 ) {
-            JsonArray biomesArray = oElem.get(biome).getAsJsonArray();
-            biomes = new ArrayList<>();
-            
-            for (JsonElement biomeEntry : biomesArray) {
-                Biome bm = ForgeRegistries.BIOMES.getValue(new ResourceLocation(biomeEntry.getAsString()));
+	private static void loadOres(JsonArray ores, DimensionBuilder builder) {
+		List<SpawnBuilder> spawns = new ArrayList<>();
+		for( JsonElement e : ores ) {
+			JsonObject ore = e.getAsJsonObject();
+			SpawnBuilder spawn = builder.newSpawnBuilder(null);
+			OreBuilder tO = spawn.newOreBuilder();
+			String blockID = ore.get("blockID").getAsString();
+			int meta = ore.has("metaData")?ore.get("metaData").getAsInt():0;
+			if( meta > 0 ) {
+				tO.setOre(blockID.toString(), meta);
+			} else {
+				tO.setOre(blockID.toString());
+			}
+			float frequency = ore.has("frequency")?ore.get("frequency").getAsFloat():20.0f;

```

Comment:
"frequency", "minHeight", "maxHeight", "variation", "default" and "size" should be shared constants